test_runner/regress/test_walredo_no_zombies.py

```diff
@@ -0,0 +1,102 @@
+from pathlib import Path
+import subprocess
+import time
+
+import pytest
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnvBuilder, PageserverApiException, PageserverHttpClient
+from fixtures.types import TenantId, TimelineId
+
+import psutil
+
+def assert_child_processes(pageserver_pid, wal_redo_present=False, defunct_present=False):
+    # XXX --ppid is probably GNU-only
+    children = list(psutil.Process(pageserver_pid).children())
+    for child in children:
+        if not wal_redo_present:
+            assert "--wal-redo" not in list(child.cmdline())
+        if not defunct_present:
+            assert child.status() != psutil.STATUS_ZOMBIE
+
+def test_walredo_no_zombies(neon_env_builder: NeonEnvBuilder):
```

Comment:
```suggestion
# Check that the pageserver doesn't leave behind WAL redo processes
# when a tenant is detached. We had an issue previously where we failed
# to wait and consume the exit code of the WAL redo process, leaving it behind
# as a zombie process.
def test_walredo_not_left_behind_on_detach(neon_env_builder: NeonEnvBuilder):
```

This actually checks that there are no WAL redo processes, zombies or alive, left behind. Which makes perfect sense, but let's reflect that in the test name.